src/Tools/dotnet-monitor/Auth/MonitorApiKeyConfigurationObserver.cs

```diff
@@ -17,24 +17,29 @@ internal class MonitorApiKeyConfigurationObserver :
     {
         private readonly ILogger<MonitorApiKeyConfigurationObserver> _logger;
         private readonly IOptionsMonitor<MonitorApiKeyConfiguration> _options;
+        private readonly IAuthConfiguration _authConfigurationOptions;
 
         private IDisposable _changeRegistration;
 
         public MonitorApiKeyConfigurationObserver(
             ILogger<MonitorApiKeyConfigurationObserver> logger,
-            IOptionsMonitor<MonitorApiKeyConfiguration> options
-            )
+            IOptionsMonitor<MonitorApiKeyConfiguration> options,
+            IAuthConfiguration authConfigurationOptions)
         {
             _logger = logger;
             _options = options;
+            _authConfigurationOptions = authConfigurationOptions;
         }
 
         public void Initialize()
         {
             _changeRegistration = _options.OnChange(OnMonitorApiKeyOptionsChanged);
 
-            // Write out current validation state of options when starting the tool.
-            CheckMonitorApiKeyOptions(_options.CurrentValue);
+            if (_authConfigurationOptions.KeyAuthenticationMode != KeyAuthenticationMode.NoAuth)
```

Comment:
Because "no auth vs auth" configuration cannot change during the lifetime of `dotnet monitor`, I would also push the `_changeRegistration = ...` line inside the condition. There's no point in monitoring for authentication changes if the tool is running in no-auth mode.